experimenter/experimenter/klaatu/client.py

```diff
@@ -0,0 +1,95 @@
+import io
+import json
+import zipfile
+
+import requests
+from django.conf import settings
+
+
+class KlaatuClient:
+    def __init__(self, workflow_name):
+        self.workflow_name = workflow_name
+        self.auth_token = settings.GITHUB_AUTH_TOKEN
+        self.base_url = "https://api.github.com/repos/mozilla/klaatu"
+        self.headers = {
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {self.auth_token}",
+            "Content-Type": "application/json",
+        }
+
+    def run_test(self, experiment_slug, branch_slug, targets):
+        url = f"{self.base_url}/actions/workflows/{self.workflow_name}/dispatches"
+
+        data = {
+            "ref": "main",
+            "inputs": {
+                "slug": experiment_slug,
+                "branch": json.dumps([branch_slug]),
+                "firefox-version": json.dumps(targets),
+            },
+        }
+
+        response = requests.post(url, headers=self.headers, json=data)
+        if response.status_code != 204:
+            raise Exception(
+                f"Failed to trigger workflow: {response.status_code}, {response.text}"
+            )
+
+    def find_run_id(self, experiment_slug, dispatched_at=None):
+        created_filter = ""
+        if dispatched_at:
+            created_filter = f"&created=>={dispatched_at.isoformat()}Z"
```

Comment:
Here's an example where we would use [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)